addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1316,13 +1314,16 @@ def _create_sub_channel(self, from_message_id=None, name=None):
     @api.readonly
     @api.model
     def get_mention_suggestions(self, search, limit=8):
-        """ Return 'limit'-first channels' id, name, channel_type and authorizedGroupFullName fields such that the
+        """ Return 'limit'-first channels' id, name, channel_type and group_public_id fields such that the
             name matches a 'search' string. Exclude channels of type chat (DM) and group.
         """
         domain = [("name", "ilike", search), ("channel_type", "=", "channel")]
         channels = self.search(domain, limit=limit)
         return [{
-            'authorizedGroupFullName': channel.group_public_id.full_name,
+            'group_public_id': {
+                'id': channel.group_public_id.id,
+                'model': 'res.groups'
+            },
```

Comment:
This is not flat, maybe we should take the opportunity the return a store result here too.